Framework/DataHandling/src/SaveGSS.cpp

```diff
@@ -92,391 +136,502 @@ void SaveGSS::init() {
+      // Add data to buffer
+      generateBankData(m_outputBuffer[specIndex], specIndex);
+      m_progress->report();
+    }
+  }
 }
 
-//--------------------------------------------------------------------------------------------
-/** Write the header information, which is independent of bank, from the given
- * workspace
-   * @param format :: The string containing the header formatting
-   * @param os :: The stream to use to write the information
-   * @param primaryflightpath :: Value for the moderator to sample distance
-   */
-void SaveGSS::writeHeaders(const std::string &format, std::stringstream &os,
-                           double primaryflightpath) const {
-  const Run &runinfo = inputWS->run();
-  std::ios::fmtflags fflags(os.flags());
+/**
+  * Creates the file header information, which should be at the top of
+  * each GSAS output file from the given workspace.
+  *
+  * @param l1 :: Value for the moderator to sample distance
+  * @return :: A string stream containing the bank header details to be
+  * written to the start of the file
+*/
+void SaveGSS::generateInstrumentHeader(std::stringstream &out,
+                                       double l1) const {
+  const Run &runinfo = m_inputWS->run();
+  const std::string format = getPropertyValue("Format");
 
   // Run number
   if (format == SLOG) {
-    os << "Sample Run: ";
-    writeLogValue(os, runinfo, "run_number");
-    os << " Vanadium Run: ";
-    writeLogValue(os, runinfo, "van_number");
-    os << " Wavelength: ";
-    writeLogValue(os, runinfo, "LambdaRequest");
-    os << "\n";
+    out << "Sample Run: ";
+    getLogValue(out, runinfo, "run_number");
+    out << " Vanadium Run: ";
+    getLogValue(out, runinfo, "van_number");
+    out << " Wavelength: ";
+    getLogValue(out, runinfo, "LambdaRequest");
+    out << "\n";
   }
 
   if (this->getProperty("ExtendedHeader")) {
     // the instrument parameter file
     if (runinfo.hasProperty("iparm_file")) {
       Kernel::Property *prop = runinfo.getProperty("iparm_file");
       if (prop != nullptr && (!prop->value().empty())) {
-        std::stringstream line;
-        line << "#Instrument parameter file: " << prop->value();
-        os << std::setw(80) << std::left << line.str() << "\n";
+        out << std::setw(80) << std::left;
+        out << "#Instrument parameter file: " << prop->value() << "\n";
       }
     }
 
-    // write out the gsas monitor counts
-    os << "Monitor: ";
+    // write out the GSAS monitor counts
+    out << "Monitor: ";
     if (runinfo.hasProperty("gsas_monitor")) {
-      writeLogValue(os, runinfo, "gsas_monitor");
+      getLogValue(out, runinfo, "gsas_monitor");
     } else {
-      writeLogValue(os, runinfo, "gd_prtn_chrg", "1");
+      getLogValue(out, runinfo, "gd_prtn_chrg", "1");
     }
-    os << "\n";
+    out << "\n";
   }
 
-  if (format == SLOG) {
-    os << "# "; // make the next line a comment
+  if (format.compare(SLOG) == 0) {
```

Comment:
https://stackoverflow.com/a/9158971
Huh yep - I guess I misread that we should use `compare` instead of `==`